### PR TITLE
fix: better parse sci notation with `+` and `-`

### LINKED ISSUE #54: invalid unit for `10e-1px`

### PR DESCRIPTION
**Pull request by evilebottnawi**
fix #54 

/cc @TrySound

### CODE REVIEW

File: test/unit.js [hunk 1]
```diff
@@ -57,6 +57,78 @@ var tests = [
...
+    fixture: "1.1e-1e",
+    expected: { number: "1.1e-1", unit: "e" }
+  },
+  {
+    fixture: "1.1e-1rem",
+    expected: { number: "1.1e-1", unit: "rem" }
+  },
+  {
+    fixture: "1.1e--++1e",
+    expected: { number: "1.1e-", unit: "-++1e" }
```
Comment (TrySound) [inline test/unit.js]: Is this valid number?

Comment (evilebottnawi) [reply test/unit.js]: @TrySound no, we already stop parsing for second `e`, so i use same logic for `+` and `-`

Comment (evilebottnawi) [reply test/unit.js]: Our fixtures contains:
```
{
    fixture: "2.",
    expected: { number: "2.", unit: "" }
  },
```

`2.` is not valid number too :smile:

**Comment by TrySound** [comment]
Yep. Looks like it's the wrong case. Don't really remember why I added it.

**Comment by evilebottnawi** [comment]
@TrySound :+1: i will send a PR

**Pull request merged**